docs/specification.md

```diff
@@ -389,4 +398,104 @@ unit
 
 **TODO**
 
+### **lock_bond**
+
+Types
+```haskell
+type Parameter =
+  | LockBond Bond
+
+-- TODO: Not exactly sure what it is
+data Bond = Bond
+  { boMutez :: Mutez
+  , boAssets :: _undefined
+  , boPercentage :: _undefined
+  }
+
+```
+
+Parameter (in Michelson):
+```
+TODO
+```
+
+- Sender MUST submit assets/tez according to bond requirement
+
+- TODO: Explain what is the "percentage associated with bond"
+```
+if there is a percentage associated with that bond.
+Bonds are required - if there’s no value, it can be set to 0.
+```
+
+### **propose**
+
+Types
+```haskell
+type Parameter =
+  | Propose Proposal
+
+data Proposal = Proposal
+  { prName :: String
+  , prDescription :: String
+  , prMetadata :: ProposalMetaData
+  , prFrozenTokenRequirement :: Int
+  , ... -- TODO: Not sure what else
+  }
+
+data ProposalMetaData = _Hole
+```
+
+Parameter (in Michelson):
+```
+TODO
+```
+
+- Sender MUST have BOND locked as specified by requirement.
+
+### **proposal_metadata**
+
+Types
+```haskell
+type Parameter =
+  | ProposalMetadata MetadataParam
+
+data MetadataParam = MetadataParam
+  { voProposalId :: Int
+  , .. -- TODO: not sure what other params are needed
+  }
+```
+
+Parameter (in Michelson):
+```
+TODO
+```
+- Return metadata of the proposal associated with the id
+- Fails with `TODO: Add error message` if the proposal id is
+not associated with any on going proposals.
+
+### **vote**
+
+Types
+```haskell
+type Parameter =
+  | Vote VoteParam
+
+data VoteParam = VoteParam
+  { voProposalId :: Int
+  , .. -- TODO: not sure what other params are needed
```

Comment:
I think there can be a positive vote and a negative vote, so at least `Bool` is needed.

Reply:
I've added `type VoteType = Upvote | Downvote` here to be more descriptive.

Reply:
Ok, but I think in Michelson it should still be `bool` because that's more efficient.